crates/turborepo-auth/src/auth/login.rs

```diff
@@ -20,13 +30,35 @@ pub async fn login<'a>(
     existing_token: Option<&'a str>,
     login_url_configuration: &str,
     login_server: &impl LoginServer,
-) -> Result<Cow<'a, str>, Error> {
-    // Check if token exists first.
+) -> Result<Token, Error> {
+    // Check if passed in token exists first.
     if let Some(token) = existing_token {
         if let Ok(response) = api_client.get_user(token).await {
             println!("{}", ui.apply(BOLD.apply_to("Existing token found!")));
             ui::print_cli_authorized(&response.user.email, ui);
-            return Ok(token.into());
+            return Ok(Token {
+                token: token.into(),
+                is_existing: true,
+            });
+        }
+    }
+
+    // If the user is logging into Vercel, check for an existing `vc` token.
+    if login_url_configuration.contains("vercel.com") {
```

Comment:
Is there anyway we could get some automated test either via a unit test or a prysk test that could test this path getting hit?